File: dart/data_structures/stack.dart
// stack.dart

class Stack {
  List<dynamic> _items = [];

  bool get isEmpty => _items.isEmpty;

  void push(dynamic item) {
    _items.add(item);
  }

  dynamic pop() {
    if (isEmpty) {
      throw StateError("Stack is empty");
    }
    return _items.removeLast();
  }

  dynamic peek() {
    if (isEmpty) {
      throw StateError("Stack is empty");
    }
    return _items.last;
  }

  void display() {
    print("Stack: $_items");
  }
}

void main() {
  Stack myStack = Stack();

  myStack.push(1);
  myStack.push(2);
  myStack.push(3);

  print("Original Stack:");
  myStack.display();

  print("\nPop operation result: ${myStack.pop()}");

  print("Stack after Pop operation:");
  myStack.display();
}

File: go/data_structures/stack.go
// stack.go

package main

import "fmt"

type Stack struct {
	items []int
}

func (s *Stack) Push(item int) {
	s.items = append(s.items, item)
}

func (s *Stack) Pop() (int, error) {
	if len(s.items) == 0 {
		return 0, fmt.Errorf("stack is empty")
	}

	index := len(s.items) - 1
	item := s.items[index]
	s.items = s.items[:index]
	return item, nil
}

func (s *Stack) IsEmpty() bool {
	return len(s.items) == 0
}

func (s *Stack) Size() int {
	return len(s.items)
}

func (s *Stack) Display() {
	fmt.Println("Stack:", s.items)
}

func main() {
	stack := &Stack{}

	stack.Push(1)
	stack.Push(2)
	stack.Push(3)

	fmt.Println("Stack Size:", stack.Size())

	item, err := stack.Pop()
	if err == nil {
		fmt.Println("Popped Item:", item)
	} else {
		fmt.Println(err)
	}

	fmt.Println("Is Stack Empty:", stack.IsEmpty())
	stack.Display()
}